google-cloud-logging/src/main/java/com/google/cloud/logging/LogEntry.java

```diff
@@ -570,7 +600,19 @@ static LogEntry fromPb(com.google.logging.v2.LogEntry entryPb) {
     builder.setLabels(entryPb.getLabelsMap());
     builder.setSeverity(Severity.fromPb(entryPb.getSeverity()));
     if (!entryPb.getLogName().equals("")) {
-      builder.setLogName(LogName.parse(entryPb.getLogName()).getLog());
+      LogName name = LogName.parse(entryPb.getLogName());
+      builder.setLogName(name.getLog());
+      ResourceName resource = ResourceName.fromLogName(name);
+      /**
+       * Limitation: we dont know if project ID was provided originally by destination parameter
+       * explicitly or was taken from credentials (since we do not serialize ResourceName object
+       * into PB payload). This mean that for project ID we assume it was taken from credentials
+       * (thus we never create destination object for project ID based path)
+       */
+      if (resource != null
+          && !resource.getOptionType().equals(ResourceName.DestinationType.PROJECT)) {
+        builder.setDestination(ResourceName.fromLogName(name));
+      }
```

Comment:
As I mentioned in comment above the code, the project ID based destination could be originally generated from ResourceName.PROJECT parameter or from project ID coming from global/credentials data. Since we do not serialize the ResourceName data into PB payload, in deserialization stage we cannot determine how the original LogEntry was created (with or without destination). Thats mean that for given LogEntry deserialize(serialize(LogEntry)) might not be equal to original LogEntry, hence the test `testToAndFromPb()` was failing on following line:
`compareLogEntry(logEntry, LogEntry.fromPb(logEntry.toPb("project")));`
I could adjust tests or I could assume that for project based destination do not create an object.
WDYT?